devito/operator.py

```diff
@@ -278,21 +279,24 @@ def _prepare_arguments(self, **kwargs):
         args = args.reduce_all()
 
         # All DiscreteFunctions should be defined on the same Grid
-        functions = [kwargs.get(p, p) for p in self.input if p.is_DiscreteFunction]
-        mapper = ReducerMap([('grid', i.grid) for i in functions if i.grid])
+        grids = {getattr(p, 'grid', None) for p in self.input} - {None}
+        if len(grids) > 1 and configuration['mpi']:
```

Comment:
in theory, we could. But there's a test you wrote time ago that would fail, the one which does spatial subsampling -- it uses two grids. Maybe we can change it ?
note that it was this same way before this PR

Reply:
Oh right. I think we would need a "parent" grid concept to makes sure everything is related.

Reply:
plausible